scripts/Muon/GUI/ElementalAnalysis/elemental_analysis.py

```diff
@@ -428,3 +447,91 @@ def minor_peaks_checked(self):
     def minor_peaks_unchecked(self):
         for element, selector in iteritems(self.element_widgets):
             self.checked_data(element, selector.secondary_checkboxes, False)
+
+    def add_line_by_type(self, run, _type):
+        # Ensure all detectors are enabled
+        for detector in self.detectors.detectors:
+            if not detector.isEnabled():
+                detector.setEnabled(True)
+
+        if self.plot_window is None:
+            return
+
+        # Plot the correct line type on all open subplots
+        color = 'C3'
+        if _type == 'Total':
+            color = 'C0'
+        elif _type == 'Prompt':
+            color = 'C1'
+        elif _type == 'Delayed':
+            color = 'C2'
+        for subplot in self.plotting.get_subplots():
+            for ws in mantid.mtd['{}; Detector {}'.format(run, subplot[-1])]:
+                if _type in ws.getName():
+                    self.plotting.plot(subplot, ws.getName(), color=color)
+
+    def remove_line_type(self, run, _type):
+        if self.plot_window is None:
+            return
+
+        # Remove the correct line type on all open subplots
+        for subplot in self.plotting.get_subplots():
+            for ws in mantid.mtd['{}; Detector {}'.format(run, subplot[-1])]:
+                if _type in ws.getName():
+                    self.plotting.remove_line(subplot, ws.getName())
+
+        # If no line type is selected do not allow plotting
+        self.uncheck_detectors_if_no_line_plotted()
+
+    def uncheck_detectors_if_no_line_plotted(self):
+        if not any([
+                self.lines.total.isChecked(),
+                self.lines.prompt.isChecked(),
+                self.lines.delayed.isChecked()
+        ]):
+            for detector in self.detectors.detectors:
+                detector.setEnabled(False)
+
+    # When removing a line with the remove window uncheck the line here
+    def uncheck_on_removed(self, removed_lines):
+        for line in removed_lines:
+            if 'Total' in line:
+                self.lines.total.blockSignals(True)
+                self.lines.total.setChecked(False)
+                self.lines.total.blockSignals(False)
+            if 'Prompt' in line:
+                self.lines.prompt.blockSignals(True)
+                self.lines.prompt.setChecked(False)
+                self.lines.prompt.blockSignals(False)
+            if 'Delayed' in line:
+                self.lines.delayed.blockSignals(True)
+                self.lines.delayed.setChecked(False)
+                self.lines.delayed.blockSignals(False)
+        self.uncheck_detectors_if_no_line_plotted()
+
+    # Line total
+    def line_total_checked(self):
```

Comment:
can pass the bool state and use that.

Reply:
should use this in the above.

Reply:
Should this be done with `electrons_checked`, `gamma_checked` etc.?

Reply:
yeah, its better to have one function instead of two